### DISCUSSION TITLE
Breaking change: EODataAccessGateway.download returns a local file path

### DISCUSSION THREAD

**Discussion by maximlt** [CS-SI/eodag#245] [General]
@apparell, @ahuarte47, @remi-braun 

We noticed that the paths returned by the download methods weren't consistent, you can find more details in this issue https://github.com/CS-SI/eodag/issues/232

https://github.com/CS-SI/eodag/pull/244 modifies `EODataAccessGateway.download` and `EOProduct.download` so that they return a local file path (as `download_all`) instead of a file URI path. This will be effective in the next version (certainly 2.3.0)

The `location` attribute of an `EOProduct` still holds the file URI path to the downloaded product (once downloaded obviously).

NB:
I'm reaching you directly through this process since you've been active on the project's GitHub. Let me know if you don't want me to pin you again here (hopefully there won't be many breaking changes :) )

**Comment by ahuarte47** [comment]
Thank you very much for the info @maximlt !

**Reply by remi-braun** [reply]
Thanks for the update !
So if I'm getting this correctly, if I was only using `download_all`, ity won't change anything for me ?

**Comment by maximlt** [comment]
Exactly Rémi

**Reply by remi-braun** [reply]
🍾
